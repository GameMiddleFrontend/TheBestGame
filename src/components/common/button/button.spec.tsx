import React from 'react';
import renderer from 'react-test-renderer';
import Button from './button';
import cancelArrowImg from '@styles/images/cancel-arrow.svg';

describe('Button', () => {
  it('ButtonText Snapshot', () => {
    const tree = renderer.create(<Button className={'button button-text'}>TextButton</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('ButtonIconOnly Snapshot', () => {
    const tree = renderer.create(<Button className={'button button-icon-only'} icon={cancelArrowImg} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
