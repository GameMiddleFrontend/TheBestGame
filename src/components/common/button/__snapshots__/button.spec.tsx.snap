// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button ButtonIconOnly Snapshot 1`] = `
<button
  className="button button button-icon-only"
  onClick={[Function]}
>
  <img
    role="img"
    src="test-file-stub"
  />
</button>
`;

exports[`Button ButtonText Snapshot 1`] = `
<button
  className="button button button-text"
  onClick={[Function]}
>
  TextButton
</button>
`;
